### PR TITLE
readme: add an introduction to the repo content

### PR DESCRIPTION
**Pull request by Julio-Guerra**
Update the readme file so that it introduces the products living in the repository.
It includes some enhancements to the other existing sections which would also need some rework but is out of the scope of this change.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,7 +1,19 @@
...
+
+This repository contains Go packages for the client-side components of the Datadog product suite for Application Performance Monitoring, Continuous Profiling and Application Security Monitoring of Go applications.
+
+- [Datadog Application Performance Monitoring (APM)](https://docs.datadoghq.com/tracing/): Trace requests as they flow across web servers, databases and microservices so that developers have great visiblity into bottlenecks and troublesome requests.  
+The package [`gopkg.in/DataDog/dd-trace-go.v1/ddtrace/tracer`](https://pkg.go.dev/gopkg.in/DataDog/dd-trace-go.v1/ddtrace/tracer) allows you to trace any piece of your Go code, and commonly used Go libraries can be automatically traced thanks to our out-of-the-box integrations which can be found in the package [`gopkg.in/DataDog/dd-trace-go.v1/ddtrace/contrib`](https://pkg.go.dev/gopkg.in/DataDog/dd-trace-go.v1/contrib).
+
+- [Datadog Go Continuous Profiler](https://docs.datadoghq.com/profiler/): Continuously profile your Go apps to find CPU, memory, and IO bottlenecks, broken down by function name, and line number, to significantly reduce end-user latency and infrastructure costs.
```
Comment (nsrip-dd) [inline README.md:12]: We don't offer IO profiling (yet), so how about synchronization instead?

```suggestion
- [Datadog Go Continuous Profiler](https://docs.datadoghq.com/profiler/): Continuously profile your Go apps to find CPU, memory, and synchronization bottlenecks, broken down by function name, and line number, to significantly reduce end-user latency and infrastructure costs.  
```

**Review by nsrip-dd — CHANGES_REQUESTED** [review]
Great idea, thanks Julio! I just have one small suggestion for the profiling item (see comment)

File: README.md [hunk 2]
```diff
@@ -1,7 +1,19 @@
...
+
+- [Datadog Go Continuous Profiler](https://docs.datadoghq.com/profiler/): Continuously profile your Go apps to find CPU, memory, and synchronization bottlenecks, broken down by function name, and line number, to significantly reduce end-user latency and infrastructure costs.  
+The package [`gopkg.in/DataDog/dd-trace-go.v1/profiler`](https://pkg.go.dev/gopkg.in/DataDog/dd-trace-go.v1/profiler) allows you to periodically collect and send Go profiles to the Datadog API.
+
+- [Datadog Application Security Monitoring (ASM)](https://docs.datadoghq.com/security_platform/application_security/): Get observability into your Go apps attacks that aim to exploit code-level vulnerabilities, such as Server-Side-Request-Forgery (SSRF), SQL injection, Log4Shell, and Reflected Cross-Site-Scripting (XSS). ASM is not a standalone Go package and is transparently integrated into the APM tracer. Simply [enable it](https://docs.datadoghq.com/security_platform/application_security/setup_and_configure/?code-lang=go) in order to make benefit from the security monitor.
```
Comment (Julio-Guerra) [inline README.md:15]:
```suggestion
- [Datadog Application Security Monitoring (ASM)](https://docs.datadoghq.com/security_platform/application_security/): Get observability into your Go apps attacks that aim to exploit code-level vulnerabilities, such as Server-Side-Request-Forgery (SSRF), SQL injection, Log4Shell, and Reflected Cross-Site-Scripting (XSS). ASM is not a standalone Go package and is transparently integrated into the APM tracer. Simply [enable it](https://docs.datadoghq.com/security_platform/application_security/setup_and_configure/?code-lang=go) to make benefit from the security monitoring.
```

**Review by nsrip-dd — DISMISSED** [review]

File: README.md [hunk 3]
```diff
@@ -11,7 +23,7 @@ go get gopkg.in/DataDog/dd-trace-go.v1/ddtrace/tracer
 go get gopkg.in/DataDog/dd-trace-go.v1/profiler
 ```
 
-Additionally there are many [contrib](./contrib) packages that can be installed as needed like this:
+Additionally there are many [contrib](./contrib) packages that can be installed to automatically instrument and trace comonly used Go libraries such as [net/http](https://pkg.go.dev/gopkg.in/DataDog/dd-trace-go.v1/contrib/net/http), [gorilla/mux](https://pkg.go.dev/gopkg.in/DataDog/dd-trace-go.v1/contrib/gorilla/mux) or [database/sql](https://pkg.go.dev/gopkg.in/DataDog/dd-trace-go.v1/contrib/database/sql):
```
Comment (ajgajg1134) [inline README.md:26]:
```suggestion
Additionally there are many [contrib](./contrib) packages that can be installed to automatically instrument and trace commonly used Go libraries such as [net/http](https://pkg.go.dev/gopkg.in/DataDog/dd-trace-go.v1/contrib/net/http), [gorilla/mux](https://pkg.go.dev/gopkg.in/DataDog/dd-trace-go.v1/contrib/gorilla/mux) or [database/sql](https://pkg.go.dev/gopkg.in/DataDog/dd-trace-go.v1/contrib/database/sql):
```

**Review by ajgajg1134 — CHANGES_REQUESTED** [review]
Just a couple small typos

**Review by ajgajg1134 — APPROVED** [review]
Looks good to me!

**Review by nsrip-dd — APPROVED** [review]

**Pull request merged**